api/packets.py

```diff
@@ -169,8 +169,8 @@ async def wrapper(packet: Packet, session: Session) -> None:
                 _type = typing.cast(str, _type)
 
                 if _type == "bytes":
-                    data[field] = bytes(packet.data)
-                    packet.data.clear()
+                    data[field] = packet.data
+                    packet.data = b""
```

Comment:
not really sure why this is here? :o
does the non-bytes case actually consume the bytes from the packet?

Reply:
yes it does, i think this needs to remain a bytearray. i’d suggest looking over how i do packet reading

Reply:
ah i see - read thru and seems fine. bytearray isn't required, as you're not modifying the buffer but you might want to use a memoryview of the bytes since you're slicing it for more efficiency.

Reply:
yeah perhaps

Reply:
that can be a separate PR - all type checks are passing btw

```sh
(venv) cmyui:~/programming/akatsuki/herbe.rt@main$ make mypy
api/bancho.py:117: error: Argument 5 to "create" has incompatible type "Optional[OsuVersion]"; expected "OsuVersion"
Found 1 error in 1 file (checked 48 source files)```

Reply:
ah nice

Reply:
i can implement the memoryview part next

Reply:
it should be done at the PacketArray level it looks like